modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaClientCodegen.java

```diff
@@ -541,6 +552,58 @@ public CodegenModel fromModel(String name, Model model, Map<String, Model> allDe
         return objs;
     }
 
+    public void preprocessSwagger(Swagger swagger) {
+        if (swagger != null && swagger.getPaths() != null) {
+            for (String pathname : swagger.getPaths().keySet()) {
+                Path path = swagger.getPath(pathname);
+                if (path.getOperations() != null) {
+                    for (Operation operation : path.getOperations()) {
+                        boolean hasFormParameters = false;
+                        for (Parameter parameter : operation.getParameters()) {
+                            parameter.getVendorExtensions().put("x-isBody", parameter instanceof BodyParameter);
+                            if (parameter instanceof FormParameter) {
+                                hasFormParameters = true;
+                            }
+                        }
+
+                        String defaultContentType = hasFormParameters ? "application/x-www-form-urlencoded" : "application/json";
+                        String contentType = operation.getConsumes() == null || operation.getConsumes().isEmpty()
+                                ? defaultContentType : operation.getConsumes().get(0);
+                        String accepts = getAccept(operation);
+                        operation.setVendorExtension("x-contentType", contentType);
+                        operation.setVendorExtension("x-accepts", accepts);
+                    }
```

Comment:
Are `x-contentType` and `x-accepts` similar to `produces` and `consumes`?

In Java (and other programming languages), we've helper methods (defined in the generated code) to pick one ContentType and Accepts from a list of MIME types. Ref: https://github.com/swagger-api/swagger-codegen/blob/master/samples/client/petstore/java/default/src/main/java/io/swagger/client/ApiClient.java#L348